### ISSUE TITLE
bug: Button type and form bindings does not work properly in v7

### ISSUE THREAD

**Issue by nstekovic** [ionic-team/ionic-framework#27952]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

v7.x

### Current Behavior

It seems that starting from v7 there's been a bug with ion-button binding. I believe this is caused by this [merge](https://github.com/ionic-team/ionic-framework/pull/27790/files#diff-8b0e185d2b3a4ceee4ec690a5f8ff2992ef3720594562f34bf2cae17f05254ffR156). When using [type] along with [form] binding for ion-button it just doesn't work. This is the case when we have a dynamic form id set. 

Having form like this:
```typescript
  form: FormGroup = new FormGroup({});
  formId = 'foo';
  buttonType = 'submit';

  handleFormSubmit(): void {
    // never triggeres it
    console.log('test');
  }
```
 
```html
<form [id]="formId" [formGroup]="form" (ngSubmit)="handleFormSubmit()"></form>
<ion-button [type]="buttonType" [form]="formId">Submit</ion-button>
```

where component is actually part of modal won't work.

Fair to mention that It had worked in version 6 just fine.

### Expected Behavior

Type and form binding work and finds form on click instead of connectedCallback.

### Steps to Reproduce

Create an app where you would just have simple button that opens modal as component. Inside component, you would have form where form id, button type and form inputs is binded dynamically.
 

### Code Reproduction URL

https://stackblitz.com/edit/angular-wrvuyk-dtvuqu?file=src%2Fapp%2Fexample.component.html,src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fapp.component.html,src%2Fapp%2Fexample.component.ts

### Ionic Info

```
Ionic:

   Ionic CLI                     : 7.1.1
   Ionic Framework               : @ionic/angular 7.2.2
   @angular-devkit/build-angular : 15.2.9
   @angular-devkit/schematics    : 15.2.9
   @angular/cli                  : 15.2.9
   @ionic/angular-toolkit        : 10.0.0

Capacitor:

   Capacitor CLI      : 5.2.2
   @capacitor/android : 5.2.2
   @capacitor/core    : 5.2.2
   @capacitor/ios     : 5.2.2

Cordova:

   Cordova CLI       : not installed
   Cordova Platforms : not available
   Cordova Plugins   : not available

Utility:

   cordova-res : not installed globally
   native-run  : 1.7.2

System:

   Android SDK Tools : 26.1.1 (/Users/nstekovic/Library/Android/sdk)
   NodeJS            : v18.17.0 (/Users/nstekovic/.nvm/versions/node/v18.17.0/bin/node)
   npm               : 9.6.7
   OS                : macOS Unknown
   Xcode             : Xcode 14.3.1 Build version 14E300c

```

**Comment by liamdebeasi** [comment]
Thanks for the report. Here is a dev build with a proposed fix if you are interested in testing:

```
7.2.3-dev.11691523847.1760ab58
```

Note that you may need to install this dev build in a project locally. StackBlitz sometimes has issue installing these dev build dependencies.

**Comment by nstekovic** [comment]
Works as expected with `7.2.3-dev.11691523847.1760ab58`. Thanks @liamdebeasi

**Comment by liamdebeasi** [comment]
Thanks for the issue. This has been resolved via https://github.com/ionic-team/ionic-framework/pull/27955, and a fix will be available in an upcoming release of Ionic Framework.

**Comment by liamdebeasi** [comment]
A fix for this was released in Ionic v7.2.3 earlier today. Thanks again for reporting this issue!
